anomalib/models/components/feature_extractors/feature_extractor.py

```diff
@@ -0,0 +1,159 @@
+        super().__init__()
+
+        # Check if argument is passed as a key word argument or as a single argument of dictionary or dataclass.
+        feature_extractor_params = self._get_feature_extractor_params(args, kwargs)
+        self.feature_extractor = self._assign_feature_extractor(feature_extractor_params)
+        self.layers = (
+            self.feature_extractor.layers
+            if isinstance(self.feature_extractor, TimmFeatureExtractor)
+            else self.feature_extractor.return_nodes
+        )
+        self._out_dims: List[int]
+
+    def _get_feature_extractor_params(self, args, kwargs):
+        """Performs validation checks and converts the arguments to the correct data type.
+
+        Checks if the arguments are passed as a key word argument or as a single argument of dictionary or dataclass.
+        If the checks pass, returns the feature extractor parameters as a dataclass.
+
+        The feature extractor expects only one of args of kwargs
+
+        Args:
+            args (Union[TimmFeatureExtractorParams, TorchFXFeatureExtractorParams, DictConfig]): Feature extractor
+                parameters.
+            kwargs (Dict[str, Any]): Feature extractor parameters as key word arguments.
+        """
+        if len(args) == 1:
+            feature_extractor_params = self._convert_datatype(args[0])
+        elif len(args) > 0 and kwargs is not None:
+            raise ValueError(
+                "Either arguments as keyword arguments or as a single argument of type TimmFeatureExtractorParams or"
+                " TorchFXFeatureExtractorParams"
+            )
+        else:
+            feature_extractor_params = self._convert_datatype(kwargs)
+        return feature_extractor_params
+
+    def _convert_datatype(
+        self,
+        feature_extractor_params: Union[TimmFeatureExtractorParams, TorchFXFeatureExtractorParams, DictConfig, Dict],
+    ):
+        """When config us loaded from entry point scripts, the data type of the arguments is DictConfig.
+
+        Args:
+            feature_extractor_params: Feature extractor parameters to convert.
+
+        Returns:
+            Union[TimmFeatureExtractorParams, TorchFXFeatureExtractorParams]: Converted feature extractor parameters.
+        """
+        if isinstance(feature_extractor_params, (DictConfig, dict)):
+            if "layers" in feature_extractor_params:
+                feature_extractor_params = TimmFeatureExtractorParams(**feature_extractor_params)
+            else:
+                feature_extractor_params = TorchFXFeatureExtractorParams(**feature_extractor_params)
+        if not isinstance(feature_extractor_params, (TimmFeatureExtractorParams, TorchFXFeatureExtractorParams)):
+            raise ValueError(f"Unknown feature extractor params type: {type(feature_extractor_params)}")
+        return feature_extractor_params
```

Comment:
If I understand correctly, the mechanism to distinguish between TorchFX and Timm depends on the occurrence of the `layers` key in the dictionary. I'm not sure if this is desirable. It might lead to confusing situations where the user thinks they're using torchvision backbone but actually it's Timm. 

How about setting the framework explicitly, by passing a `framework` parameter that defaults to `torchvision`?